### PR TITLE
Fix broken link in documentation

### PR DESCRIPTION
**Pull request by annagrin**
### Description

Fix broken documentation link that block PRs (see failing tests in https://github.com/NVIDIA/cuda-quantum/pull/1590)

### CODE REVIEW

**Review by khalatepradnya — APPROVED** [review]
👍🏽

File: docs/sphinx/using/backends/simulators.rst [hunk 1]
```diff
@@ -205,7 +205,7 @@ Use the following commands to enable distribution across multiple GPUs (adjust t
 .. note::
 
   If the `CUTENSORNET_COMM_LIB` environment variable is not set, MPI parallelization on the :code:`tensornet` backend may fail.
-  If you are using a CUDA Quantum container, this variable is pre-configured and no additional setup is needed. If you are customizing your installation or have built CUDA Quantum from source, please follow the instructions for `activating the distributed interface <https://docs.nvidia.com/cuda/cuquantum/latest/cutensornet/getting_started.html#install-cutensornet-from-nvidia-devzone>`__ for the `cuTensorNet` library. This requires 
+  If you are using a CUDA Quantum container, this variable is pre-configured and no additional setup is needed. If you are customizing your installation or have built CUDA Quantum from source, please follow the instructions for `activating the distributed interface <https://docs.nvidia.com/cuda/cuquantum/latest/getting-started/index.html#from-nvidia-devzone>`__ for the `cuTensorNet` library. This requires
```
Comment (khalatepradnya) [inline docs/sphinx/using/backends/simulators.rst:208]: Not sure if it's nature of rendering or the space after `requires` is missing. Please check.

**Comment by schweitzpgi** [comment]
Is this a duplicate?

**Comment by khalatepradnya** [comment]
> Is this a duplicate?

This is targeting `main` branch... this fix is required for all the CI pipelines.

**Comment by schweitzpgi** [comment]
> > Is this a duplicate?
> 
> This is targeting `main` branch... this fix is required for all the CI pipelines.

Ah, I see it now. Thanks, Pradnya.

File: docs/sphinx/using/backends/simulators.rst (hunk 1 above)
Comment (annagrin) [reply docs/sphinx/using/backends/simulators.rst]: Thanks, done!

**Pull request merged**